web/modules/custom/ai_screening_project/src/Helper/ProjectHelper.php

```diff
@@ -268,13 +271,130 @@ public static function getSubscribedEvents(): array {
+      $optionsGroupContributors = $this->mapUsersToSelectOptions($users);
+
+      $form['group_fieldset'] = [
+        '#type' => 'fieldset',
+        '#title' => t('Project group'),
+      ];
+
+      $form['group_fieldset']['groupOwnerSelect'] = [
+        '#type' => 'select',
+        '#title' => t('Project owner'),
+        '#default_value' => $groupOwnerId,
+        '#options' => $optionsGroupOwner,
+        '#attributes' => ['class' => ['text-full form-text required bg-primary text-primary border border-primary rounded-md py-2 px-3 my-1 w-full']],
+        '#weight' => 1,
+      ];
+
+      $form['group_fieldset']['groupUsersSelect'] = [
+        '#type' => 'select',
+        '#title' => t('Contributors'),
+        '#description' => t('Which users are allowed to contribute to this project'),
+        '#description_display' => 'before',
+        '#options' => $optionsGroupContributors,
+        '#multiple' => TRUE,
+        '#default_value' => array_keys($optionsGroupOwner),
+        '#attributes' => ['class' => ['use-choicesjs-plugin bg-primary text-primary border border-primary rounded-md py-2 px-3 my-1 w-full']],
+        '#weight' => 2,
+      ];
+
+      $form['#validate'][] = [$this, 'validateGroupsForm'];
+      $form['actions']['submit']['#submit'][] = [$this, 'submitGroupsForm'];
+    }
+  }
+
+  /**
+   * Custom validation for group part of form.
+   */
+  public function validateGroupsForm(array &$form, FormStateInterface $form_state): void {
+    if (!in_array($form_state->getValue('groupOwnerSelect'), $form_state->getValue('groupUsersSelect'))) {
+      $form_state->setErrorByName('groupOwnerSelect', $this->t('Project owner must be a contributor.'));
+    }
+  }
+
+  /**
+   * Submit groups stuff in project edit.
+   */
+  public function submitGroupsForm(array $form, FormStateInterface $formState) {
+    $group = $this->loadProjectGroup($formState->getFormObject()->getEntity());
+
+    // Add/remove members of group.
+    $groupUserIds = array_keys($this->mapUsersToSelectOptions($group->getRelatedEntities('group_membership')));
+    $selectedGroupContributors = $formState->getValue('groupUsersSelect');
+    $memberToAdd = array_diff($selectedGroupContributors, $groupUserIds);
+    $memberToRemove = array_diff($groupUserIds, $selectedGroupContributors);
+
+    foreach ($memberToAdd as $userId) {
+      $user = $this->userStorage->load($userId);
+      $group->addMember($user);
+    }
+
+    foreach ($memberToRemove as $userId) {
+      $user = $this->userStorage->load($userId);
+      $group->removeMember($user);
+    }
+
+    // Change group owner.
+    $groupOwnerForm = $formState->getValue('groupOwnerSelect');
+    $currentGroupOwner = $group->getOwner()->id();
+
+    // @todo validate that the group owner is in selectedGroupContributors
+    if ($groupOwnerForm !== $currentGroupOwner) {
+      $groupOwnerId = (int) $groupOwnerForm;
+      $group->setOwner($this->userStorage->load($groupOwnerId));
```

Comment:
You don't need the `$currentGroupOwner` variable and (probably) not the cast:

```suggestion
    $groupOwner = $formState->getValue('groupOwnerSelect');

    // @todo validate that the group owner is in selectedGroupContributors
    if ($groupOwner !== $group->getOwner()->id()) {
      $group->setOwner($this->userStorage->load($groupOwner));
```